index.js

```diff
@@ -52,4 +52,112 @@ const state = {
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+const store = document.querySelector(".store--item-list")
+const cart = document.querySelector(".cart--item-list")
+
+
+function renderStoreItems() {
+  store.innerHTML= ""
+
+  state.items.forEach((item) => {
+    const li = document.createElement("li")
+    store.appendChild(li)
+    const div = document.createElement("div")
+    div.setAttribute("class","store--item-icon")
+    li.appendChild(div)
+    const image = document.createElement("img")
+    image.src = `assets/icons/${item.id}.svg`
+    image.alt = `${item.id}`
+    div.appendChild(image)
+    const button = document.createElement("button")
+    button.innerText = "Add to cart";
+    li.appendChild(button)
+    button.addEventListener("click", (event)=>{
+      addItemToCart(item)
+      renderCart()
+      cartTotal()
+    
+    }
+    )
+  })
+
+}
+
+
+function renderCart (){
+  cart.innerHTML= ""
+
+  state.cart.forEach((cartItem) => {
+    const cartLi = document.createElement("li")
+    cart.appendChild(cartLi)
+    const cartImg = document.createElement("img")
+    cartImg.src = `assets/icons/${cartItem.id}.svg`
+    cartImg.alt = `${cartItem.id}`
+    cartLi.appendChild(cartImg)
+    const itemName = document.createElement('p')
+    itemName.innerText = `${cartItem.name}`
+    cartLi.appendChild(itemName)
+    const plusButton = document.createElement("button")
+    const span = document.createElement("span")
+    const minusButton = document.createElement("button")
+    minusButton.setAttribute("class","quantity-btn remove-btn center")
+    span.setAttribute("class","quantity-text center")
+    plusButton.setAttribute("class","quantity-btn add-btn center")
+    plusButton.innerText="+"
+    span.innerText=`${cartItem.quantity}`
+    minusButton.innerText="-"
+    cartLi.appendChild(plusButton)
+    cartLi.appendChild(span)
+    cartLi.appendChild(minusButton)
+   
+    plusButton.addEventListener("click",(event)=>{
+      cartItem.quantity += 1
+      renderCart()
+      cartTotal()
+    })
+    
+    
+    minusButton.addEventListener("click",(event)=>{
+      cartItem.quantity -= 1
+      
+      if(cartItem.quantity === 0){
+        const index = state.cart.indexOf(cartItem)
+        state.cart.splice(index, 1)
+      }
+      renderCart()
+      cartTotal()
+    })
+  })
+}
+
+
+function addItemToCart(item){
+  
+   if( state.cart.find((produce) => produce.name === item.name ) === undefined){
+    item.quantity = 1
+    state.cart.push(item)
+   } else {
+   
+    item.quantity += 1
+   }
+ 
+  }
+  
+  function cartTotal (){
+    const total = document.querySelector(".total-number")
+    let totalPrice = 0
+    state.cart.forEach((item)=>{
+     totalPrice +=  item.price * item.quantity
+
+    })
+     span
+    const formatCurrency = new Intl.NumberFormat('en-GB', {
```

Comment:
Nice Sam, I havent ever seen this before, could you please explain how it works?